File: DogeFriendsSharedClassLibrary/Dto/UserInfoDto.cs
namespace DogeFriendsSharedClassLibrary.Dto
{
    public class UserInfoDto
    {
        public int Id { get; set; }
        public string Username { get; set; } = string.Empty;
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string Showname { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string? Hometown { get; set; }
        public string? Description { get; set; }
        public string ExternalId { get; set; } = string.Empty;
    }
}

File: AngularDogeFriendsClient/src/app/models/UserInfoDto.ts
import { IImage } from "./Images";

export interface UserInfoDto {
  id: number;
  username: string;
  firstName: string;
  lastName: string;
  showname: string;
  email: string;
  hometown: string | null;
  description: string | null;
  externalId: string;
}

export interface UserInfoDtoWithMainImage extends UserInfoDto {
  mainImage?: string | null;
}

